### PR TITLE
Separate gcp timer but without the timer

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Notes
This was supposed to be a Draft PR - please don't merge :)

Alternative to #1799 that uses a service restart every 5 minutes instead of a timer. This might be frowned upon.
The nice(?) thing about it is that the recurring job is controlled entirely by creating/deleting the marker file - no need to call `systemctl stop ua-license-check.timer` like in the timer approach.

Look at [this commit](https://github.com/canonical/ubuntu-advantage-client/pull/1801/commits/a3540b327967ef9774de1c7b948b9b88008c3dde) to see the code diff from #1799

## Proposed Commit Message

> summary: no more than 70 characters
>
> A description of what the change being made is and why it is being
> made, if the summary line is insufficient.  The blank line above is
> required. This should be wrapped at 72 characters, but otherwise has
> no particular length requirements.
>
> If you need to write multiple paragraphs, feel free.
>
> LP: #NNNNNNN (replace with the appropriate Launchpad bug reference if applicable)
> Fixes: #NNNNNNN (replace with the appropriate github issue if applicable)

## Desired commit type::

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [ ] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist:

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

File: systemd/ua-license-check.service [hunk 1]
```diff
@@ -0,0 +1,11 @@
+[Unit]
+Description=Ubuntu Advantage check for GCP Ubuntu Pro licenses
+After=network.target network-online.target systemd-networkd.service ua-auto-attach.service
```
Comment (lucasmoura) [inline systemd/ua-license-check.service:3]: Why do we need to specify all of these extra targets here ? Wouldn't waiting for `ua-auto-attach.service`  enough to run this service ?

Comment (blackboxsw) [reply systemd/ua-license-check.service:3]: @lucasmoura auto-attach service is only present on PRO images, in the absence of that service or target, we need to fallback to the next earliest time in boot we need to wait for.... which is that network is "up" on the system so we can talk to the IMDS.

**Review by blackboxsw — CHANGES_REQUESTED** [review]
I'd prefer your alternative #1799 because it feels like the Reboot=always RebootSec=5mins is creating a "timer-like" object that is not represented in `systemctl list-timers` and therefore less discoverable. I'll let others weigh in though if they have strong opinions on this.

File: systemd/ua-license-check.service (hunk 1 above)
Comment (lucasmoura) [reply systemd/ua-license-check.service:3]: Ohh excelent point.

**Comment by orndorffgrant** [comment]
closing in favor of #1799

**Pull request closed without merging**